### PR TITLE
Fix DanglingLine boundary flows computation

### PR DESCRIPTION
**Pull request by caioluke**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

Currently when computing the flows (P & Q) at boundary side in iIDM, the shunt admittance is split between the terminal and the boundary side for DanglingLine-s. This is incorrect because does not match the DanglingLine model where the shunt admittance is only on network side, and also causes issues such as the ones described in https://github.com/powsybl/powsybl-core/pull/3165.

**What is the new behavior (if this is a feature change)?**
Shunt admittance for DanglingLine is not considered as split on both sides anymore when computing the boundary P and Q flows.

**Does this PR introduce a breaking change or deprecate an API?**
- [X] No

### CODE REVIEW

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineBoundaryImpl.java
Comment (jeandemanged) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineBoundaryImpl.java:1]: please adjust the existing comment at the top of the class as follows:

```diff
-    // for SV use: side represents the network side, that is always
-   // Side.ONE for a dangling line.
+   // Notes about SV utility class usage here:
+   // - side represents the network side, which is always Side.ONE for a dangling line.
+   // - DanglingLine model has shunt admittance on network side only, hence splitShuntAdmittance argument in SV methods must be set to false (when not set, SV considers it as false).
```

**Review by jeandemanged — CHANGES_REQUESTED** [review]

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineBoundaryImpl.java
Comment (jeandemanged) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineBoundaryImpl.java:1]: ho and I was still looking at #3165 in my review ... tou need to adjust the SV calls here!

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java [hunk 1]
```diff
@@ -47,23 +43,19 @@ public DanglingLineData(DanglingLine danglingLine, boolean splitShuntAdmittance)
             return;
         }
 
-        double g1 = splitShuntAdmittance ? danglingLine.getG() * 0.5 : danglingLine.getG();
-        double b1 = splitShuntAdmittance ? danglingLine.getB() * 0.5 : danglingLine.getB();
-        double g2 = splitShuntAdmittance ? danglingLine.getG() * 0.5 : 0.0;
-        double b2 = splitShuntAdmittance ? danglingLine.getB() * 0.5 : 0.0;
-
```
Comment (olperr1) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java:54]: Can you also add a comment in this method?

```java
// DanglingLine model has shunt admittance on network side only, so it is not split between both sides.
```

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractMergeNetworkTest.java [hunk 2]
```diff
@@ -127,10 +127,10 @@ public void testMergeAndDetach() {
...
         DanglingLine dl2 = merge.getDanglingLine("dl2");
         // - P0 and Q0 of the removed tie line's underlying dangling lines were updated:
-        assertEquals(-1724.437, dl1.getP0(), 0.001);
-        assertEquals(1605.281, dl1.getQ0(), 0.001);
-        assertEquals(-1724.437, dl2.getP0(), 0.001);
-        assertEquals(1605.281, dl2.getQ0(), 0.001);
+        assertEquals(-731.312, dl1.getP0(), 0.001);
+        assertEquals(-1254.625, dl1.getQ0(), 0.001);
+        assertEquals(-731.312, dl2.getP0(), 0.001);
+        assertEquals(-1254.625, dl2.getQ0(), 0.001);
```
Comment (jeandemanged) [inline iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractMergeNetworkTest.java:133]: note: this test case uses a dummy network with _huge_ B and G shunt values for dangling lines (G = 4.0 Siemens and B = 5.0 Siemens) hence the large diff in results. The values used are not representative of what exists in real life which would be rather in the range of _eventually up to_ a milli-Siemens for _very_ long overhead lines or cables.

File: iidm/iidm-serde/src/test/resources/V1_0/tieline.xml [hunk 3]
```diff
@@ -37,6 +37,6 @@
...
+    <iidm:tieLine id="NHV1_NHV2_1" ucteXnodeCode="X1" bus1="NHV1" connectableBus1="NHV1" voltageLevelId1="VLHV1" bus2="NHV2" connectableBus2="NHV2" voltageLevelId2="VLHV2" p1="302.4440612792969" q1="98.74027252197266" p2="-300.43389892578125" q2="-137.18849182128906" id_1="NHV1_NHV2_1.1" r_1="1.5" x_1="16.5" g1_1="0.0" b1_1="9.65E-5" g2_1="0.0" b2_1="9.65E-5" xnodeP_1="-301.43898434342725" xnodeQ_1="-118.89616773372641" id_2="NHV1_NHV2_1.2" r_2="1.5" x_2="16.5" g1_2="0.0" b1_2="9.65E-5" g2_2="0.0" b2_2="9.65E-5" xnodeP_2="301.43897577958876" xnodeQ_2="118.89616087695104"/>
+    <iidm:tieLine id="NHV1_NHV2_2" ucteXnodeCode="X2" bus1="NHV1" connectableBus1="NHV1" voltageLevelId1="VLHV1" bus2="NHV2" connectableBus2="NHV2" voltageLevelId2="VLHV2" p1="302.4440612792969" q1="98.74027252197266" p2="-300.43389892578125" q2="-137.18849182128906" id_1="NHV1_NHV2_2.1" r_1="1.5" x_1="16.5" g1_1="0.0" b1_1="9.65E-5" g2_1="0.0" b2_1="9.65E-5" xnodeP_1="-301.43898434342725" xnodeQ_1="-118.89616773372641" id_2="NHV1_NHV2_2.2" r_2="1.5" x_2="16.5" g1_2="0.0" b1_2="9.65E-5" g2_2="0.0" b2_2="9.65E-5" xnodeP_2="301.43897577958876" xnodeQ_2="118.89616087695104"/>
```
Comment (jeandemanged) [inline iidm/iidm-serde/src/test/resources/V1_0/tieline.xml:41]: here we can see nicely the bug fix, it seems this network has a good powerflow solution (bus voltages and angles), and we see here it had bad (unbalanced) boundary data, which are now fixed.

**Before** the fix:
- the difference boundary P was xnodeP_1="-301.47434650169697" vs xnodeP_2="301.472319625184", hence 2 digits precision.
- the difference boundary Q was xnodeQ_1="-118.8505833038079" vs xnodeQ_2="118.76748245598334", hence 0 (!) digits precision.

**After** the fix:
- on P: xnodeP_1="-301.43898434342725" and xnodeP_2="301.43897577958876" hence 4 digits precision (+2 !)
- on Q: xnodeQ_1="-118.89616773372641" and xnodeQ_2="118.89616087695104" hence 5 digits precision (+3 !)

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java [hunk 4]
```diff
@@ -47,23 +43,20 @@ public DanglingLineData(DanglingLine danglingLine, boolean splitShuntAdmittance)
...
         Complex vBoundaryBus = new Complex(Double.NaN, Double.NaN);
         if (danglingLine.getP0() == 0.0 && danglingLine.getQ0() == 0.0) {
-            LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(danglingLine.getR(), danglingLine.getX(), 1.0, 0.0, 1.0, 0.0, new Complex(g1, b1), new Complex(g2, b2));
+            LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(
+                    danglingLine.getR(), danglingLine.getX(), 1.0, 0.0, 1.0, 0.0, new Complex(danglingLine.getG(), danglingLine.getB()), new Complex(0, 0));
             vBoundaryBus = adm.y21().multiply(v1).negate().divide(adm.y22());
         } else {
 
             // Two buses Loadflow
             Complex sBoundary = new Complex(-danglingLine.getP0(), -danglingLine.getQ0());
```
Comment (jeandemanged) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java:57]: This code has an issue where is considers only the DanglingLine load part and ignores the optional generation part. This would need to be fixed in another PR.

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java [hunk 5]
```diff
@@ -47,23 +43,20 @@ public DanglingLineData(DanglingLine danglingLine, boolean splitShuntAdmittance)
...
             Complex ytr = new Complex(danglingLine.getR(), danglingLine.getX()).reciprocal();
-            Complex ysh2 = new Complex(g2, b2);
+            Complex ysh2 = new Complex(0, 0);
             Complex zt = ytr.add(ysh2).reciprocal();
             Complex v0 = ytr.multiply(v1).divide(ytr.add(ysh2));
             double v02 = v0.abs() * v0.abs();
```
Comment (jeandemanged) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java:62]: `ysh` is now zero there is no point adding it and other simplifications should show up

**Review by jeandemanged — CHANGES_REQUESTED** [review]

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java (hunk 5 above)
Comment (jeandemanged) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java]: see 155378be6655df4ef1291e3e13799a4eb528182b

**Review by olperr1 — APPROVED** [review]
